### DISCUSSION TITLE
Any way to pause the game?

### DISCUSSION THREAD

**Discussion by Dorbmon** [scripthookvdotnet/scripthookvdotnet#1295] [Q&A] [answered]
I want to gather some data, and it may need a long time. I dont want any change in game during that time. So can I pause the game? By the way, the native function SET_GAME_PAUSED seems not working...

**Comment by Dorbmon** [comment]
And is there any way to get the tasks?
Like below a lot of task hints:
![image](https://github.com/scripthookvdotnet/scripthookvdotnet/assets/17807931/ee55d916-fd0b-49ae-88b5-c96e76f01494)

**Reply by kagikn** [reply]
Sorry, but I have no idea how you can get cached log strings, such as mission objectives, scripted conversations, or notification (aka feed) texts. I haven't researched how the game cache these strings.

**Reply by Dorbmon** [reply]
Oh, so sad. I may take some research about this.

**Comment by kagikn** [comment, marked as answer]
To pause the game by opening the frontend menu, try `ACTIVATE_FRONTEND_MENU, like this:
```cs
Function.Call(Hash.ACTIVATE_FRONTEND_MENU, Game.GenerateHash("FE_MENU_VERSION_SP_PAUSE"), true /* bPauseGame */, -1 /* HighlightTab (unique id), an invalid value will open the default tab */);
```
The issue is, SHV scripts can't run tick functions at all when you pause the game by pressing the pause button or calling `ACTIVATE_FRONTEND_MENU` with the 2nd argument true (keyboard callbacks actually run however). I think you would have to give up gathering game stuff while the game is paused...

You can pause the game without the menu with `Game.Pause(bool)` but do note that **ALL** SHVDN scripts run while the game is paused by that method.

**Reply by Dorbmon** [reply]
Really thank you!!!
